File: _733_flood_fill/main.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        if (image[sr][sc] == color)
        {
            return image;
        }

        int current = image[sr][sc];
        image[sr][sc] = color;

        // Up
        if (sr >= 1 && current == image[sr - 1][sc])
        {
            floodFill(image, sr - 1, sc, color);
        }

        // Down
        if (sr < image.size() - 1 && current == image[sr + 1][sc])
        {
            floodFill(image, sr + 1, sc, color);
        }

        // Left
        if (sc >= 1 && current == image[sr][sc - 1])
        {
            floodFill(image, sr, sc - 1, color);
        }

        // Right
        if (image.size() > 0 && sc < image[0].size() - 1 && current == image[sr][sc + 1])
        {
            floodFill(image, sr, sc + 1, color);
        }
        
        return image;
    }
};

File: _733_flood_fill/main.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        if image[sr][sc] == color:
            return image

        oldColor = image[sr][sc]
        image[sr][sc] = color

        if (sr >= 1 and oldColor == image[sr - 1][sc]):
            self.floodFill(image, sr - 1, sc, color)

        if (sr < len(image) - 1 and oldColor == image[sr + 1][sc]):
            self.floodFill(image, sr + 1, sc, color)

        if (sc >= 1 and oldColor == image[sr][sc - 1]):
            self.floodFill(image, sr, sc - 1, color)

        if (len(image) > 0 and sc < len(image[0]) - 1 and oldColor == image[sr][sc + 1]):
            self.floodFill(image, sr, sc + 1, color)

        return image
